### DISCUSSION TITLE
Filecoin最新挖矿硬件配置和集群方案

### DISCUSSION THREAD

**Discussion by jyma** [filecoin-project/community-china#18] [矿机配置]
# Filecoin最新挖矿硬件配置和集群方案

## 1. 机器配置总览
[FilGuard团队](https://github.com/filguard) 基于AMD7542 CPU 和 RTX 3070/3080/3090显卡，测试出一套性价比高、封装速度快、简单实用的挖矿集群硬件配置，该方案有如下特点：
- PreCommit(P1, P2, C1)、Commit(C2)分别由单独的2台Worker完成；
- 可根据机器的具体配置(单路、双路、远程C2)，自由灵活的搭配两种机器的比例；
- 搭配FilGuard团队优化后的软件程序，可达到平均每台机器封装32GB扇区**每天3.2T**、64GB扇区**每天1.6T**的封装速度；
- 搭配FilGuard团队的分布式Miner程序，可解决Miner单点风险、及显卡冲突等问题；

#### 1.1 Daemon推荐配置
|产品名称|型号|数量|备注|
|------|-----------------------|:-----:|---------|
| 主板 | 超微 H12SSL-i | 1 ||
| CPU | AMD 宵龙 EPYC 7252/7282 | 1 ||
| 散热器 | AMD SP3-2U 纯铜 | 1 ||
| 内存 | 三星 DDR4-2933 ECC REG 64GB | 4 ||
| 网卡 | Intel X520-SR2 万兆网卡 + 万兆多模光模块 * 2 | 1 ||
| 系统盘 | 三星 M.2 512GB | 1 ||
| 数据盘 | 三星 U.2 3.84TB | 1 ||
| 转接卡 | PCIe转U.2转接卡（支持2U机箱） + 转接线 |1||
| 机箱 | 2U机箱 + 暴力风扇 + 800W电源 | 1 ||

#### 1.2 Seal-Miner配置
|产品名称|型号|数量|备注|
|------|-----------------------|:-----:|---------|
| 主板 | 超微 H12SSL-i | 1 ||
| CPU | AMD 宵龙 EPYC 7252/7282  | 1 ||
| 散热器 | AMD SP3-2U 纯铜  | 1 ||
| 内存 | 三星 DDR4-2933 ECC REG 64GB | 4 ||
| 系统盘 | 三星 M.2 512GB  | 1 ||
| 网卡 | Intel 40G网卡 双口 | 1 ||
| 光模块 | Intel 40G网卡双口 + 40G多模光模块 * 2 | 1 ||
| 机箱 | 2U机箱 + 暴力风扇 + 800W电源  | 1 ||

#### 1.3 PoSt-Miner配置
|产品名称|型号|数量|备注|
|------|-----------------------|:-----:|---------|
| 主板 | 超微 H12SSL-i | 1 ||
| CPU | AMD 宵龙 EPYC 7F32| 1 ||
| 散热器 | AMD SP3-2U 纯铜 | 1 ||
| 内存 | 三星 DDR4-2933 ECC REG 64GB | 4 ||
| 显卡| RTX 3080 / 3090 涡轮公版  | 2 ||
| 网卡 | Intel 40G网卡双口 + 40G多模光模块 * 2 | 1 ||
| 系统盘 | 三星 M.2 512GB | 1 ||
| 机箱电源 | 4U机箱 + 暴力风扇 + 1300W电源 | 1 ||

#### 1.4 PreCommit Worker配置
|产品名称|型号|数量|备注|
|------|-----------------------|:-----:|---------|
| 主板 | 技嘉 MZ32-ARO  | 1 ||
| CPU | AMD 宵龙 EPYC 7542 | 1 ||
| 散热器 | AMD SP3-2U 纯铜 | 1 ||
| 内存 | 三星 DDR4-2933 ECC REG 64GB | 16 ||
| 显卡| RTX 3070 涡轮 技嘉/微星 | 1 ||
| 网卡 | Intel X520-SR2 万兆网卡 + 万兆多模光模块 * 2 | 1 ||
| 系统盘 | 三星 M.2 256GB | 1 ||
| 缓存盘 | 三星 U.2 3.84TB + 转接线(直插主板) | 4 ||
| 机箱 | 4U机箱 + 暴力风扇 + 800W电源 | 1 ||

#### 1.5 Commit Worker配置
|产品名称|型号|数量|备注|
|------|-----------------------|:-----:|---------|
| 主板 | 超微 H12SSL-i | 1 ||
| CPU | AMD 宵龙 EPYC 7F32 | 1 ||
| 散热器 | AMD SP3-2U 纯铜 | 1 ||
| 内存 | 三星 DDR4-2933 ECC REG 64GB | 8 ||
| 显卡 | RTX 3090 涡轮公版 | 2 ||
| 网卡 | Intel X520-SR2 万兆网卡 + 万兆多模光模块 * 2 | 1 ||
| 系统盘 | 三星 M.2 512GB | 1 ||
| 机箱 | 4U机箱 + 暴力风扇 + 1300W电源 | 1 ||

## 2. 集群方案
总算力小于10PB，日封装100TB左右的挖矿集群，都可以基于此方案进行搭建。
### 2.1 集群架构
无论是小集群、还是中大规模的集群，在网络上都要做到计算集群和存储集群的分离，实现网络隔离。

整个集群划分为管理集群、封装Miner集群、计算集群、存储集群、证明Miner集群，各个子集群作用如下：
- 管理集群：跳板机和管理机，可以用一台机器，也可以分开，跳板机上要做好安全策略和登录白名单，最好使用硬件进行登录认证。
- 封装Miner集群：包含Seal-Miner(封装)、Deal-Miner(接单)以及Daemon，用于管理计算节点，给计算Worker派发任务，完成密封结果数据的落盘。
- 计算集群：包含PreCommit Worker和Commit Worker，完成密封计算过程。
- 存储集群：NFS、ZFS、Ceph或其他商业存储（七牛云、阿里云、华为云）。
- 证明Miner集群：Window-PoSt-Miner(负责时空证明)和Winning-PoSt-Miner(负责爆块)，还包含一台时空证明和爆块的备用机。

网络拓扑如下图所示（点击图片可在新窗口中查看大图）：

- 2个华为 AR-6140H-S 万兆路由器，分别连接1条上联线路，实现负载均衡（如果只有单条线路，可以只用1个路由器，没有冗余）；
- 2个华为 CE6820 48口万兆交换机交换机使用堆叠线做堆叠；
- 1个Seal-Miner、2个PoSt-Miner通过双40G端口(bond0 负载均衡模式)，接入交换机的40G端口；
- Daemon、Worker、Monitor、Storage通过双10G端口(bond0 负载均衡模式)，接入交换机10G端口；

![5PB集群网络拓扑](https://user-images.githubusercontent.com/1715211/126285892-83035f14-c339-436d-84d9-cfe5342ca5c8.png)

### 2.2 Worker配比说明
按照以上机器配置，Worker机器配比如下：
|PreCommit Worker显卡|Commit Worker显卡|比例|备注|
|------|-----------------------|:-----:|---------|
| NVIDIA 3070 * 1 | NVIDIA 3080 * 2 | 3 : 2 ||
| NVIDIA 3070 * 1 | NVIDIA 3080 * 3 | 2 : 1 ||
| NVIDIA 3070 * 1 | NVIDIA 3090 * 2 | 2 : 1 ||

**Comment by perfectyc** [comment]
大佬，带我啊

**Reply by 466919153** [reply]
水水水水

**Reply by KaixuanZhao** [reply]
IPFS主网最大雅典娜矿池，提供一站式采购托管运营，fil存量数据迁移出海等服务，全网算力领先1400PB，需要B端1PB以上，C端50T，100T，200T（可扩容）和FIL存量数据迁移出海服务方案的滴滴我，欢迎加入社群Wechet：kkmmjjnn00

**Reply by linjun826668** [reply]
我想问一下，如果不托管在IDC机房。网速和IP的要求是多少呢

**Comment by icedtea6** [comment]
招聘啊 有没有技术负责人来一个   1021979490@qq.com

**Reply by maxvint** [reply]
老板，招聘在技术招聘版块发呀，记得附上薪资哦，哈哈哈

**Reply by icedtea6** [reply]
> 老板，招聘在技术招聘版块发呀，记得附上薪资哦，哈哈哈

加我私聊呗

**Reply by lhjok** [reply]
> 老板，招聘在技术招聘版块发呀，记得附上薪资哦，哈哈哈

需要传输这么大量的数据吗？全部万兆口，还有4万兆，还要双口，有必要吗？

**Reply by lhjok** [reply]
这么高的传输速度，你的机械硬盘有这么快吗？

**Reply by lhjok** [reply]
居然还要万兆路由器，请问万兆的宽带哪里有卖？

**Comment by hyithack** [comment]
大佬厉害呀,希望得到您的垂青!感谢!

**Reply by icedtea6** [reply]
> 大佬厉害呀,希望得到您的垂青!感谢!

搞安全的你是？

**Comment by qqggkkaa0101** [comment]
想自己组建一个小集群  不太懂搭建 想请教您

**Reply by maxvint** [reply]
集群搭建可以按照论坛里面的文章，有问题欢迎提问呀，论坛里好多大佬。

**Reply by dachengcheng2022** [reply]
> 集群搭建可以按照论坛里面的文章，有问题欢迎提问呀，论坛里好多大佬。

什么论坛啊 给一个链接

**Reply by hackhy** [reply]
> 集群搭建可以按照论坛里面的文章，有问题欢迎提问呀，论坛里好多大佬。

哪个论坛呀，同问

**Comment by qqggkkaa0101** [comment]
微信liyivip



在 2021-03-16 18:14:10，"icedtea6" ***@***.***> 写道：

可以加个联系方式细聊




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "filecoin-project/community-china" ***@***.***&gt;;
发送时间:&nbsp;2021年3月16日(星期二) 凌晨5:03
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [filecoin-project/community-china] Filecoin最新挖矿集群硬件配置 (#18)






想自己组建一个小集群 不太懂搭建 想请教您

**Comment by qqggkkaa0101** [comment]
微信liyivip











在 2021-03-16 18:14:10，"icedtea6" ***@***.***> 写道：

可以加个联系方式细聊




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "filecoin-project/community-china" ***@***.***&gt;;
发送时间:&nbsp;2021年3月16日(星期二) 凌晨5:03
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [filecoin-project/community-china] Filecoin最新挖矿集群硬件配置 (#18)






想自己组建一个小集群 不太懂搭建 想请教您

**Comment by irocn** [comment]
大佬们，加入filecoin优化群！
<img src="https://www.irocn.cn/static/media/uploads/391616119712_.pic.jpg" width="20%" />

**Reply by y905699146** [reply]
失效了

**Reply by dafo374** [reply]
大佬求带，想进群

**Reply by spark8899** [reply]
大佬求带

**Reply by Naturefir** [reply]
求拉，vx：18202807502

**Reply by differs** [reply]
> 大佬们，加入filecoin优化群！
> <img alt="" width="20%" src="https://camo.githubusercontent.com/d30061ff8ce32897eaa22526b594d71c88cc645a2c67e9223e6e54ce61cab2c5/68747470733a2f2f7777772e69726f636e2e636e2f7374617469632f6d656469612f75706c6f6164732f3339313631363131393731325f2e7069632e6a7067">

加个群：
vx： chatid8146701ae86

**Reply by gongchengdashao** [reply]
微信群还有吗 求拉

**Reply by GZpowerleader** [reply]
广东地区矿机软硬一体方案找我18589289061

**Reply by huahuaaixuexi** [reply]
求大佬拉我进群   vx：15090432857

**Reply by zhaobo443** [reply]
大佬加个群V:zhaobo9308

**Comment by Fatman13** [comment]
求更。

**Reply by royedia** [reply]
求更

**Comment by winlong** [comment]
需要5台机器？ 平均每台5-8万元，估计要40万左右吧？不明白 1个CPU为什么要4U机箱？ 机房都是恒温的，还怕发热吗？

**Reply by ljh8mg1018** [reply]
怕顯卡放不下

**Comment by KaixuanZhao** [comment]
挖fil可以找我，VX18686935540

**Comment by dcshy** [comment]
fil集群 独立节点 机房 矿机+vx IPFS-Shy

**Comment by hayleypeng** [comment]
挖矿可以加我，vx:412853860

**Comment by KaixuanZhao** [comment]
RRMine，雅典娜矿池，全网算力领先1000PB，需要B端C端方案的滴滴我，交流学习，欢迎加入社群Wechet：kkmmjjnn00

**Comment by dcshy** [comment]
专注Filecoin，自有技术团队近百人，t3+IDC机房，日封装能力2p+，B端独立节点集群方案。可出海，微信：IPFS-Shy

**Comment by Chenhongwe** [comment]
江浙沪区域需要FIl算力封装，或者算力机租赁的联系我 15088695255同V

**Comment by zivyuan** [comment]
PreCommit Worker 的缓存盘为什么不用转接卡, 而是线? 是录错了么?

**Reply by maxvint** [reply]
如果是技嘉MZ32主板，直接用转接线插主板效果最好，不需要转接卡。

**Comment by dcshy** [comment]
Filecoin海外封装 微信：IPFS-Shy

**Comment by Chenhongwe** [comment]
单卡单U单台封装效率7T，软硬一体方案，详询v15088695255

**Comment by AidanKR** [comment]
How to SETUP  Seal Miner &  post miner??

**Comment by GZpowerleader** [comment]
广东地区矿机软硬一体方案找我18589289061

**Comment by Chenhongwe** [comment]
有的加V1113003652



在 2021-10-11 15:46:21，"yuxan liu" ***@***.***> 写道：

微信群还有吗 求拉

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by dcshy** [comment]
filecoin国内存量迁移出海 增量海外机房上架封装 独立节点集群方案 欢迎合作交流 V:IPFS-Shy

**Comment by AllynSheep** [comment]
大佬，好像没有放存储集群的配置表呢
？

**Comment by zhaobo443** [comment]
FIL联合封装，用户提供质押币，公司提供设备和运维，收益60％,公司扣出20％服务费和20%设备费，合同期1080天，540天质押币退还，t3天封装。可实地考察。V:zhaobo9308

**Reply by beck-8** [reply]
你他X的有毛病吧
